handle_actions.go

```diff
@@ -33,68 +33,34 @@ func (m *Mock) DoActions(c *Context) error {
 }
 
 func (m *Mock) doAction(c *Context, a Action) (err error) {
-	var action string
-
+	actualAction := a.GetActualAction()
 	defer func() {
 		if err != nil {
 			logrus.WithFields(logrus.Fields{
 				"err":    err,
-				"action": action,
+				"action": reflect.TypeOf(actualAction).String(),
```

Comment:
you can use `%T` to get the type in fmt package. https://golang.org/pkg/fmt/

```
fmt.Sprintf("%T", actualAction)
```

Also, avoid using reflect if possible.

Reply:
done 👍